### ISSUE TITLE
Write to tensor array fails within while loop

### ISSUE THREAD

**Issue by rays2pix** [tensorflow/tensorflow#17127]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
 Yes I am writing a while loop that reads and also updates the tensor array that's being passed to it.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
Binary
- **TensorFlow version (use command below)**:
1.3
- **Python version**: 
3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
When I attempt to write to a tensor array within a while loop, it leads me to an error 
"Could not write to TensorArray index 0 because it has already been read."

### Source code 

```
def tf_soft_nms(boxes,scores,nms_t,dt_t):

  sigma = tf.constant(0.3)
  y1 = tf.gather(boxes,0,axis=1)
  x1 = tf.gather(boxes,1,axis=1)
  y2 = tf.gather(boxes,2,axis=1)
  x2 = tf.gather(boxes,3,axis=1)
  area = tf.multiply((y2-y1) , (x2-x1))


  picked_ixs = tf.TensorArray(tf.int32,clear_after_read=False,size=0,dynamic_size=True)
  scores_ta = tf.TensorArray(scores.dtype,clear_after_read=False,size=scores.shape[0],colocate_with_first_write_call=False)
  scores_ta = scores_ta.unstack(scores)
  loop_vars = (picked_ixs,scores_ta,nms_t,dt_t)
  score_p = tf.Print(scores,[scores],"all_Scores")

  def cond(picked_ixs,scores,nms_t,dt_t):
    scores_tf = scores.stack()
    valid_indexes = tf.greater(scores_tf,dt_t)
    valid = tf.Print(valid_indexes,[valid_indexes],"Conditional check")
    #scores = scores.write(0,tf.constant(0.0))
    return tf.reduce_any(valid_indexes)

  def nms_iter(picked_ixs,scores_ta,nms_t,dt_t):
    global area
    global boxes
    #sort scores pick top score
    scores_tf = scores_ip.stack()
    scores_ixs = tf.nn.top_k(scores_tf,tf.shape(scores_tf)[0]).indices
    pick = scores_ixs[0]


    #write top score to picked
    write_index = picked_ixs.size()
    picked_ixs = picked_ixs.write(write_index,pick)
    scores_ta = scores_ta.write(pick,tf.constant(0.2))
    picked_tf = picked_ixs.stack()
     return picked_ixs,scores_ta,nms_t,dt_t

```

**Comment by rays2pix** [comment]
Adding the stack trace,

> Caused by op 'while/TensorArrayWrite_1/TensorArrayWriteV3', defined at:
>   File "nms.py", line 175, in <module>
>     test_pynms()
>   File "nms.py", line 171, in test_pynms
>     picked_tf = tf_soft_nms(boxes_tf,scores_tf,nms_t_tf,dt_tf)
>   File "nms.py", line 107, in tf_soft_nms
>     picked = tf.while_loop(cond,nms_iter,loop_vars,back_prop=False,parallel_iterations=1)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2816, in while_loop
>     result = loop_context.BuildLoop(cond, body, loop_vars, shape_invariants)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2640, in BuildLoop
>     pred, body, original_loop_vars, loop_vars, shape_invariants)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/control_flow_ops.py", line 2590, in _BuildLoop
>     body_result = body(*packed_vars_for_body)
>   File "nms.py", line 82, in nms_iter
>     scores_ta = scores_ta.write(pick,tf.constant(0.2))
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/util/tf_should_use.py", line 107, in wrapped
>     return _add_should_use_warning(fn(*args, **kwargs))
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/tensor_array_ops.py", line 310, in write
>     name=name)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 5038, in _tensor_array_write_v3
>     flow_in=flow_in, name=name)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
>     op_def=op_def)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
>     op_def=op_def)
>   File "/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
>     self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access
> 
> InvalidArgumentError (see above for traceback): TensorArray TensorArray_1_1: Could not write to TensorArray index 0 because it has already been read.
> 	 [[Node: while/TensorArrayWrite_1/TensorArrayWriteV3 = TensorArrayWriteV3[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"](while/TensorArrayStack/TensorArraySizeV3/Enter, while/strided_slice_1, while/Const_1, while/Switch_1:1)]]
>

**Comment by bignamehyp** [comment]
Please clarify where while_loop is used.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting Response: It has been 14 days with no activityand the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by drpngx** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by Han-sok** [comment]
Yes. I am experiencing exactly the same problem. I can write into the TensorArray while I am trying to write values into it in the tf.while_loop could you check this bug again? Maybe it's realted to CUDA 10.0. It would be grateful if op posts his environments.

**Comment by WolfNiu** [comment]
> Yes. I am experiencing exactly the same problem. I can write into the TensorArray while I am trying to write values into it in the tf.while_loop could you check this bug again? Maybe it's realted to CUDA 10.0. It would be grateful if op posts his environments.

Same problem happens under CUDA 9 too with TF 1.12
